src/Dropdown.elm

```diff
@@ -420,42 +449,43 @@ update (Config config) msg (State state) data =
                                 _ ->
                                     True
 
-                        focusedItem =
+                        maybeFocusedItem =
                             data
                                 |> List.indexedMap (\i item -> ( i, item ))
                                 |> List.filter (\( i, _ ) -> i == state.focusedIndex)
                                 |> List.head
                                 |> Maybe.map Tuple.second
 
-                        ( cmd, newSelectedItem ) =
+                        cmd =
                             case key of
                                 Enter ->
-                                    ( Task.succeed focusedItem
-                                        |> Task.perform config.onSelectMsg
-                                    , case focusedItem of
-                                        Nothing ->
-                                            []
-
-                                        Just item ->
-                                            [ item ]
-                                    )
-
+                                    let selectedItems_new = case maybeFocusedItem of
+                                            Just focusedItem -> modifySelectedItems focusedItem selectedItems
+                                            Nothing -> selectedItems
+                                    in
+                                        updateSelectedItemsCommand selectedItems_new
                                 _ ->
-                                    ( Cmd.none, state.selectedItems )
+                                    Cmd.none
                     in
-                    ( { state | selectedItems = newSelectedItem, focusedIndex = newIndex, isOpen = isOpen }, cmd )
+                    ( { state | focusedIndex = newIndex, isOpen = isOpen }, cmd )
     in
     ( State newState, newCommand )
 
 
 {-| Render the view
 
-    Dropdown.view dropdownConfig model.dropdownState model.items
+    Dropdown.view dropdownConfig model model.dropdownState model.items
 
 -}
-view : Config item msg -> State item -> List item -> Element msg
-view (Config config) (State state) data =
+view : Config item msg model -> model -> State -> List item -> Element msg
```

Comment:
The model has been supplied as an extra parameter, as with update function. An alternative would have been to to supply the selected Item(s) directly in the view and update functions like has been done for the list of possible options to choose from. The disadvantages of that are that you wouldn’t be able to vary the type of the selected option(s) (Maybe item/List item) depending on the type of dropdown, and also the selected item(s) has to be specified twice independently which creates the possibilities for inconsistencies.

Reply:
Now that model has been added to the type signatures of both update and view functions, it would be possible in a follow up PR to remove the extra parameter specifying the list of options from the update and view functions and instead supply the parameter in the config as a function which gives the list of options from the model, which could be nice. It could also mean the view and update type signatures are more future proof.

Reply:
So, I'm not sure which way it's better to be honest, because if we think about, the list of options and the state it's something we are already duplicating in the `view` and in the `update` functions. So, I think if we go for the approach  you mentioned, we should also refactor those 2 fields to use the same approach.

My other concern about it is the use of `model` as the generic key, a lot of components/libraries used to call the internal state as model, so I was wondering about maybe instead of use `model` we could call it `parentModel`, to be even more explicit, what do you think?

So, before making the decision about this approach, I would like to have @miguelcoba's opinion here as well

Reply:
I definitely agree that, with the approach of using a model parameter in the view function, it makes sense to remove the parameter giving the list of options. I'm not sure whether the same should apply to the State parameter, just because it's nice for State to be separate in the update function as it's a return parameter there, so I think for consistency it's nice to separate it out everywhere. Also while the list of options does feel naturally a part of the config for a dropdown, the state feels less so.

That would make the type signatures:
`view : Config item msg model -> model -> State -> Element msg
update: Config item msg model -> Msg item -> model -> State -> (State, Cmd msg)`

Reply:
>My other concern about it is the use of model as the generic key, a lot of components/libraries used to call the internal state as model, so I was wondering about maybe instead of use model we could call it parentModel, to be even more explicit, what do you think?

I wouldn't mind to call it parentModel. I did initially find the use of model parameter name confusing, but in the end it seems to be the same relationship as between msg and Msg, as between model and State (which could have been called Model), and looking at it like that made it less confusing for me.